### ISSUE TITLE
@types/node: net.Socket is incompatible with ReadableStream

### ISSUE THREAD

**Issue by robey** [DefinitelyTyped/DefinitelyTyped#11994]
```
src/test/test_linestream.ts(87,40): error TS2345: Argument of type 'Socket' is not assignable to parameter of type 'ReadableStream'.
  Types of property 'setEncoding' are incompatible.
    Type '(encoding?: string | undefined) => void' is not assignable to type '(encoding: string | null) => void'.
      Types of parameters 'encoding' and 'encoding' are incompatible.
        Type 'string | null' is not assignable to type 'string | undefined'.
          Type 'null' is not assignable to type 'string | undefined'.
```

The error description is so good that I can see how to fix it, so I'm just going to attach a pull request. The `setEncoding` on line 283 needs to be converted to the "?" form, since typescript treats null and undefined as different types.

**Comment by robey** [comment]
hmmm actually... your repo has a number of differences from the one you published as "@types/node" just a few days ago.

maybe you just need to do another release?

in fact, it looks like someone fixed this bug *more* than a few days ago, so i think something is wrong with the way you're doing releases, and they're diverging somehow.

for now, i'm going to remove the "@types/node" dependency in my project, and just commit a patched version of node.d.ts directly.

**Comment by robey** [comment]
ok i'm moderately surprised that nobody cares about this... is `@types/node` old & deprecated?

**Comment by DanielRosenwasser** [comment]
@robey the branch that needs to be fixed up is `types-2.0`, but you're probably looking at `node` on `master`. Right now we've been accepting PRs into `master` and merging them into `types-2.0`, but that's a pretty manual process.

**Comment by nikeee** [comment]
I still hade this issue to this day. Turns out that @types/node `<= 8.x.x` was affected. Using `^11.0.0` fixes the issue. Maybe we can close this issue.
